autogpt_platform/frontend/src/lib/autogpt-server-api/baseClient.ts

```diff
@@ -249,36 +249,56 @@ export default class BaseAutoGPTServerAPI {
     const hasRequestBody = method !== "GET" && payload !== undefined;
     const response = await fetch(url, {
       method,
-      headers: hasRequestBody
-        ? {
-            "Content-Type": "application/json",
-            Authorization: token ? `Bearer ${token}` : "",
-          }
-        : {
-            Authorization: token ? `Bearer ${token}` : "",
-          },
+      headers: {
+        ...(hasRequestBody && { "Content-Type": "application/json" }),
+        ...(token && { Authorization: `Bearer ${token}` }),
+      },
       body: hasRequestBody ? JSON.stringify(payload) : undefined,
     });
-    const response_data = await response.json();
 
     if (!response.ok) {
       console.warn(
         `${method} ${path} returned non-OK response:`,
-        response_data.detail,
-        response,
+        response.status,
+        response.statusText,
```

Comment:
?

Reply:
afaik `response_data.detail == response.statusText` for non-ok responses. I've changed it to print `response` only - that contains everything and `const response_data = await response.json();` isn't needed.